tools/gulp/tasks/examples.ts

```diff
@@ -0,0 +1,247 @@
+function buildImportsTemplate(metadata: ExampleMetadata): string {
+  const components = [metadata.component];
+
+  if (metadata.additionalComponents) {
+    components.push(...metadata.additionalComponents);
+  }
+
+  return `import {${components.join(',')}} from '${metadata.filename}';
+`;
+}
+
+/**
+ * Build the examples template
+ * 
+ * @param {ExampleMetadata} metadata 
+ * @returns {string} template
+ */
+function buildExamplesTemplate(metadata: ExampleMetadata): string {
+  const addlFiles = metadata.additionalFiles ? 
+    JSON.stringify(metadata.additionalFiles) : 'undefined';
+
+  const selectorName = metadata.selectorName ? 
+    ("'" + metadata.selectorName.join(', ') + "'") : 'undefined';
+
+  return `'${metadata.id}': {
+    title: '${metadata.title}',
+    component: ${metadata.component},
+    additionalFiles: ${addlFiles},
+    selectorName: ${selectorName}
+  },
+  `;
+}
+
+/**
+ * Build the list template
+ * 
+ * @param {ExampleMetadata} metadata 
+ * @returns {string} 
+ */
+function buildListTemplate(metadata: ExampleMetadata): string {
+  const components = [metadata.component];
+
+  if (metadata.additionalComponents) {
+    components.push(...metadata.additionalComponents);
+  }
+  return `${components.join(',')},
+  `;
+}
+
+/**
+ * Builds the template for the examples module
+ * 
+ * @param {ExampleMetadata[]} extractedMetadata 
+ * @returns {string} resulting template
+ */
+function buildTemplate(extractedMetadata: ExampleMetadata[]): string {
+  return `
+import {NgModule} from '@angular/core';
+import {FormsModule, ReactiveFormsModule} from '@angular/forms';
+import {CommonModule} from '@angular/common';
+import {ExampleMaterialModule} from './material-module';
+
+export interface LiveExample {
+  title: string;
+  component: any;
+  additionalFiles?: string[];
+  selectorName?: string;
+}
+
+${extractedMetadata.map(r => buildImportsTemplate(r)).join('').trim()}
+
+export const EXAMPLE_COMPONENTS = {
+  ${extractedMetadata.map(r => buildExamplesTemplate(r)).join('').trim()}
+};
+
+export const EXAMPLE_LIST = [
+  ${extractedMetadata.map(r => buildListTemplate(r)).join('').trim()}
+];
+
+@NgModule({
+  declarations: EXAMPLE_LIST,
+  entryComponents: EXAMPLE_LIST,
+  imports: [
+    ExampleMaterialModule,
+    FormsModule,
+    ReactiveFormsModule,
+    CommonModule
+  ]
+})
+export class ExampleModule { }
+`;
+}
+
+/**
+ * Given a string that is a camel or pascal case,
+ * this function will convert to dash case.
+ * 
+ * @param {string} name 
+ * @returns {string} 
+ */
+function convertToDashCase(name: string): string {
+  name = name.replace(/[A-Z]/g, ' $&');
+  name = name.toLowerCase().trim();
+  return name.split(' ').join('-')
+}
+
+/**
+ * Parse the AST of a file and get metadata about it
+ * 
+ * @param {string} filename 
+ * @param {string} src 
+ * @returns {{ primary: any, seconds: any[] }} 
+ */
+function createMetas(filename: string, src: string): { primaryComponent: any, secondaryComponents: any[] } {
```

Comment:
@DevVersion - I tinkered with tightening up the types in the `visit` function but had no luck.

I started off w/ `const visit = (node: ts.Node)` then when I hit `ClassDeclaration` I casted to that type, however, `jsDoc` is not a type on that node. I looked up the types and `jsDoc` is actually not a type on anything that extends `ts.Node` (odd?).

Also when I casted it, the `decorator.expression.expression.text` was no longer accessible. I tried the methods called `getText()` like we talked about but all returned `TypeError: Cannot read property 'text' of undefined`.